src/main/scala/org/broadinstitute/dsde/firecloud/service/RegisterService.scala

```diff
@@ -53,10 +53,17 @@ class RegisterService(val rawlsDao: RawlsDAO, val samDao: SamDAO, val thurloeDao
     }
   }
 
+  def generateWelcomeEmail(userInfo: UserInfo): Notification = {
+    userInfo match {
+      case UserInfo(_, _, _, _, None) if userInfo.isB2C => AzurePreviewActivationNotification(WorkbenchUserId(userInfo.id))
+      case _ => ActivationNotification(WorkbenchUserId(userInfo.id))
+    }
+  }
```

Comment:
I'll add code comments to this block because I'm realizing it's vague.

Essentially, we need to distinguish between the Azure-user case and the non-Azure-user case. The way to do that is to test if the user is using B2C, and then to look for a google access token (which is what the `None` is matching). If it's B2C and there's no Google access token, it's an Azure user. All other cases just use the standard activation notification.